app/frontend/src/features/search/useSearchFilters.ts

```diff
@@ -0,0 +1,94 @@
+import { Location } from "history";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { useHistory, useLocation } from "react-router-dom";
+import { UserSearchFilters as ServiceUserSearchFilters } from "service/search";
+
+export interface SearchFilters extends ServiceUserSearchFilters {
+  location?: string;
+}
+
+export default function useSearchFilters(route: string) {
+  const location = useLocation();
+  const [active, setActive] = useState<SearchFilters>(() =>
+    locationToFilters(location)
+  );
+  const pending = useRef<SearchFilters>(active);
+
+  const history = useHistory();
+
+  useEffect(() => {
+    const entries: [string, string][] = [];
+    Object.entries(active).forEach((entry) => {
+      if (typeof entry[1] === "object") {
+        Object.values(entry[1]).forEach((value) => {
+          entries.push([entry[0], `${value}`]);
+        });
```

Comment:
```suggestion
    Object.entries(active).forEach(([key, filterValues]) => {
      if (Array.isArray(filterValues)) {
        filterValues.forEach((value) => {
          entries.push([key, `${value}`]);
        });
```
Based off the type, looks like the only thing that's an "object" is `hostingStatusOptions` so the Array check is more accurate and also means you don't have to wrap it around `Object.values`